includes/admin/class-senseilms-plugin-updater.php

```diff
@@ -0,0 +1,232 @@
+ *
+ * It's a fallback to SenseiLMS_Licensing\SenseiLMS_Plugin_Updater from Sensei Pro. So it works when
+ * Sensei Pro is in a legacy version without this class.
+ *
+ * @since $$next-version$$
+ */
+class SenseiLMS_Plugin_Updater {
+
+	const CACHE_KEY_PREFIX = 'senseilms_plugin_updater_info__';
+	const CACHE_TTL        = 3600;
+
+	/**
+	 * Full qualified name for the plugin. Relative path from plugins directory. Example: 'akismet/akismet.php'.
+	 *
+	 * @var string
+	 */
+	private $plugin_full_name;
+
+	/**
+	 * Plugin slug name. Example: 'akismet'.
+	 *
+	 * @var string
+	 */
+	private $plugin_slug;
+
+	/**
+	 * Current version for the plugin. Example: '1.0.0'.
+	 *
+	 * @var string
+	 */
+	private $version;
+
+	/**
+	 * Private constructor.
+	 *
+	 * @param string $main_plugin_file_absolute_path Plugin's main file path.
+	 * @param string $version                        Plugin version.
+	 */
+	private function __construct( $main_plugin_file_absolute_path, $version ) {
+		$this->plugin_full_name = plugin_basename( $main_plugin_file_absolute_path );
+		$this->plugin_slug      = basename( $main_plugin_file_absolute_path, '.php' );
+		$this->version          = $version;
+	}
+
+	/**
+	 * Initialize the plugin updater.
+	 */
+	public static function init() {
+		add_action( 'plugins_loaded', [ __CLASS__, 'plugins_loaded' ] );
+	}
+
+	/**
+	 * Add hooks after the `plugins_loaded`, so we make sure Sensei Pro was
+	 * already loaded.
+	 */
+	public static function plugins_loaded() {
+		// Early return if license and updates are managed by WooCommerce.
+		if ( defined( 'SENSEI_COMPAT_PLUGIN' ) && SENSEI_COMPAT_PLUGIN ) {
+			return;
+		}
+
+		if ( class_exists( '\Sensei_Pro_Interactive_Blocks\Setup_Context' ) ) {
+			$instance = new self( SENSEI_IB_PLUGIN_FILE, SENSEI_IB_VERSION );
+		} elseif ( class_exists( '\Sensei_Pro\Setup_Context' ) ) {
+			$instance = new self( SENSEI_PRO_PLUGIN_FILE, SENSEI_PRO_VERSION );
+		} else {
+			return;
+		}
+
+		add_filter( 'plugins_api', [ $instance, 'get_plugin_info' ], 15, 3 );
+		add_filter( 'site_transient_update_plugins', [ $instance, 'maybe_inject_custom_update_to_update_plugins_transient' ], 15 );
+	}
+
+	/**
+	 * Get plugin information as expected by the `plugins_api` hook.
+	 * This will be called to display the details for the updated in the detailed view.
+	 *
+	 * @param false|object|array $res    Result. As defined per the `plugins_api` hook.
+	 * @param string             $action The action being executed. As defined per the `plugins_api` hook.
+	 * @param object             $args   The arguments. As defined per the `plugins_api` hook.
+	 *
+	 * @hooked plugins_api
+	 *
+	 * @return false|object If other than false is returned the actual call to wordpress.org is not done.
+	 */
+	public function get_plugin_info( $res, $action, $args ) {
+		if (
+			'plugin_information' !== $action
+			|| $this->plugin_slug !== $args->slug
+			|| false !== $res
```

Comment:
If `$res` is already set, it means that Sensei Pro already handled it. So we don't need to do it again.

Reply:
I think if `$res` isn't `false`, it was already filtered by some code, adding the Sensei Pro information there. So I think we could consider that the data is already there (even if it's not that filter you mentioned). Does it make sense to you?